pkg/lib/config/configsource/kubernetes.go

```diff
@@ -91,6 +91,7 @@ func (k *Kubernetes) Open() error {
 	secretCtrl := k.newController(corev1.ResourceSecrets, &corev1.Secret{}, k.onUpdate, k.onDelete)
 	go configMapCtrl.Run(done)
 	go secretCtrl.Run(done)
+	go k.cleanupCache(done)
```

Comment:
Consider using [this library](https://github.com/allegro/bigcache) instead? It has max number of entries, max cache size in bytes support.

Reply:
It require byte slices as value, which make the current behavior (fetch at most once, even if there are concurrent requests) impossible.